.github/workflows/docs.yml

```diff
@@ -0,0 +1,52 @@
+name: Javadocs
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
```

Comment:
do we want this on pull requests?

Reply:
Hmm, I didn't think it was problematic. This way we could catch errors in the CI / javadocs before it hits `main`. 🤷🏼

Reply:
so, each PR could push a new version of the docs, even if not getting merged?

Reply:
To prevent abuse, we can remove it.

Reply:
Oh and also to test that this even worked. Without it on PRs, we wouldn't know that the pipeline works 😄

Reply:
One idea here might be to add in an [action like this](https://github.com/marketplace/actions/pull-request-comment-trigger). Then we can "manually" trigger the docs CI on a per-PR basis.